operator_ui/src/pages/JobsIndex/JobsIndex.tsx

```diff
@@ -0,0 +1,252 @@
+import React from 'react'
+import Button from 'components/Button'
+import { Title } from 'components/Title'
+import Content from 'components/Content'
+import BaseLink from 'components/BaseLink'
+import { v2 } from 'api'
+import * as models from 'core/store/models'
+import {
+  Grid,
+  Card,
+  CardContent,
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableRow,
+  Typography,
+  TextField,
+} from '@material-ui/core'
+import Link from 'components/Link'
+import { useErrorHandler } from 'hooks/useErrorHandler'
+import { useLoadingPlaceholder } from 'hooks/useLoadingPlaceholder'
+import {
+  createStyles,
+  withStyles,
+  WithStyles,
+  Theme,
+} from '@material-ui/core/styles'
+import { DirectRequestRow } from './DirectRequestRow'
+import { OcrJobRow } from './OcrJobRow'
+import SearchIcon from '@material-ui/icons/Search'
+
+enum JobSpecTypes {
+  jobSpec = 'specs',
+  ocrJobSpec = 'jobSpecV2s',
+}
+
+interface Job<T> {
+  attributes: T
+  id: string
+  type: string
+}
+
+export type DirectRequest = Job<models.JobSpec>
+export type OffChainReporting = Job<models.OcrJobSpec>
+type CombinedJobs = DirectRequest | OffChainReporting
+
+function isDirectRequest(job: CombinedJobs): job is DirectRequest {
+  return job.type === JobSpecTypes.jobSpec
+}
+
+function isOffChainReporting(job: CombinedJobs): job is OffChainReporting {
+  return job.type === JobSpecTypes.ocrJobSpec
+}
+
+function getCreatedAt(job: CombinedJobs) {
+  if (isDirectRequest(job)) {
+    return job.attributes.createdAt
+  } else if (isOffChainReporting(job)) {
+    return job.attributes.offChainReportingOracleSpec.createdAt
+  } else {
+    return new Date().toString()
+  }
+}
+
+const PAGE_SIZE = 1000 // We intentionally set this to a very high number to avoid pagination
+
+async function getJobs() {
+  return Promise.all([
+    v2.specs.getJobSpecs(1, PAGE_SIZE),
+    v2.ocrSpecs.getJobSpecs(),
+  ]).then(([jobs, ocrJobs]) => {
+    const combinedJobs = [...jobs.data, ...ocrJobs.data]
+    const jobsByDate = combinedJobs.sort((a, b) => {
+      const jobA = new Date(getCreatedAt(a)).getTime()
+      const jobB = new Date(getCreatedAt(b)).getTime()
+      return jobA > jobB ? -1 : 1
+    })
+
+    return jobsByDate
+  })
+}
+
+// Simple Job filter. TODO: @deividas how do you feel about update this with a better object traversal functionality?
```

Comment:
Not sure what you mean by better object traversal functionality, but I agree that this will need an improvement eventually 🙂